docs/developer/plugins.md

```diff
@@ -108,16 +108,49 @@ You can then set this option in your config file:
 
 ```groovy
 // dot syntax
-myplugin.create.message = "I'm alive!"
+myplugin.createMessage = "I'm alive!"
 
-// closure syntax
+// block syntax
 myplugin {
-    create {
-        message = "I'm alive!"
+    createMessage = "I'm alive!"
+}
+```
+
+:::{versionadded} 25.02.0-edge
+:::
+
+Plugins can declare their config options by implementing the `ConfigScope` interface and declaring each config option as a field with the `@ConfigOption` annotation:
+
+```groovy
+class MyPluginConfig implements ConfigScope {
+
+    MyPluginConfig() {}
+
+    MyPluginConfig(Map opts) {
+        this.createMessage = opts.createMessage
     }
+
+    @Override
+    String name() {
+        return 'myplugin'
+    }
+
+    @Override
+    String description() {
+        return '''
+            The `myplugin` scope...
+            '''
+    }
+
+    @ConfigOption('''
+        Message to print to standard output when a run is initialized.
+        ''')
+    String createMessage
 }
```

Comment:
Here is an example of how to declare config options for the schema checker. I want to make sure we're solid on this interface before we release it to plugin developers.

I was thinking it might be better to declare the scope name/description with annotations (e.g. `@Name` and `@Description`) instead of methods. That way the schema checker doesn't need to create an instance of each class, and then you don't need the no-arg constructor.

Similarly for config options, maybe better to provide the description in a separate `@Description` annotation...